daffodil-core/src/main/scala/org/apache/daffodil/dsom/RawCommonRuntimeValuedPropertiesMixin.scala

```diff
@@ -21,25 +21,25 @@ import org.apache.daffodil.schema.annotation.props.PropertyMixin
 
 trait RawCommonRuntimeValuedPropertiesMixin
   extends PropertyMixin {
-  protected final lazy val optionByteOrderRaw = findPropertyOption("byteOrder")
+  protected final lazy val optionByteOrderRaw = findPropertyOption("byteOrder", expressionAllowed = true)
   protected final lazy val byteOrderRaw = requireProperty(optionByteOrderRaw)
-  protected final lazy val optionEncodingRaw = findPropertyOption("encoding")
+  protected final lazy val optionEncodingRaw = findPropertyOption("encoding", expressionAllowed = true)
   protected final lazy val encodingRaw = requireProperty(optionEncodingRaw)
-  protected final lazy val optionOutputNewLineRaw = findPropertyOption("outputNewLine")
+  protected final lazy val optionOutputNewLineRaw = findPropertyOption("outputNewLine", expressionAllowed = true)
   protected final lazy val outputNewLineRaw = requireProperty(optionOutputNewLineRaw)
 
-  protected final lazy val optionFillByteRaw = findPropertyOption("fillByte")
+  protected final lazy val optionFillByteRaw = findPropertyOption("fillByte", expressionAllowed = true)
   protected final lazy val fillByteRaw = requireProperty(optionFillByteRaw)
 }
 
 trait RawDelimitedRuntimeValuedPropertiesMixin
   extends RawCommonRuntimeValuedPropertiesMixin {
 
-  protected final lazy val optionInitiatorRaw = findPropertyOption("initiator")
+  protected final lazy val optionInitiatorRaw = findPropertyOption("initiator", expressionAllowed = true)
   protected final lazy val initiatorRaw = requireProperty(optionInitiatorRaw)
-  protected final lazy val optionTerminatorRaw = findPropertyOption("terminator")
+  protected final lazy val optionTerminatorRaw = findPropertyOption("terminator", expressionAllowed = true)
   protected final lazy val terminatorRaw = requireProperty(optionTerminatorRaw)
-  protected final lazy val optionChoiceLengthRaw = findPropertyOption("choiceLength")
+  protected final lazy val optionChoiceLengthRaw = findPropertyOption("choiceLength", expressionAllowed = true)
```

Comment:
choiceLength also does not allow expressions:

https://daffodil.apache.org/docs/dfdl/#_Toc398030783